packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,167 @@
+    -   `action` - for the action button/link
+-   _CSS Classes and CSS Custom Properties that affect the component_
+
+We only formally support spans of text in the `title` and default slots, but we will not explicitly prevent other HTML from being slotted there. The `action` slot only expects `nimble-button` or `nimble-anchor`, and we will use the CSS `::slotted()` pseudo-element to remove all but those two element types. We will not explicitly prevent multiple elements from being slotted in the `action` slot, though we formally only support one.
+
+### Anatomy
+
+```html
+<div class="icon">
+    ${when(x => x.severity === 'error', html`<nimble-icon-exclamation-mark
+    ></nimble-icon-exclamation-mark>`)} ${when(x => x.severity === 'warning',
+    html`<nimble-icon-triangle-filled></nimble-icon-triangle-filled>`)} ${when(x
+    => x.severity === 'info', html`<nimble-icon-info></nimble-icon-info>`)}
+</div>
+<div class="text">
+    <slot name="title"></slot>
+    <slot></slot>
+</div>
+<div class="end">
+    <slot name="action"></slot>
+    <div class="close">
+        ${when(x => !x.preventDismiss), html`
+        <nimble-button appearance="ghost" content-hidden>
+            <nimble-icon-xmark slot="start"></nimble-icon-xmark> </nimble-button
+        >` )}
+    </div>
+</div>
+```
+
+### Angular integration
+
+An Angular wrapper will be created for `nimble-banner`. No `ControlValueAccessor` is needed.
+
+### Blazor integration
+
+Blazor integration for the `nimble-banner` will be provided.
+
+### Visual Appearance
+
+See XD document link at the top of this document.
+
+The biggest issue is that we will have to completely re-style the icons, link, and buttons:
+
+-   Primary icon (error, warning, or info) color is transparent white (in all themes)
+-   Action button height is 24px (instead of 32px)
+-   Action button hover border color is white (in all themes)
+-   Action link color is white (in all themes)
+-   Close button is 16px square
+-   Close button icon is 8px square
+-   Close button hover effect is background color change (transparent white) rather than border
+
+We are not able to directly style some of these elements, because they are in the shadow DOM of `nimble-icon-*` and `nimble-button`. We will have to resort to overriding Nimble token values, like `--ni-nimble-icon-color` and `--ni-nimble-icon-size`.
+
+---
+
+## Implementation
+
+### States
+
+N/A
+
+### Accessibility
+
+_Consider the accessibility of the component, including:_
+
+-   _Keyboard Navigation and Focus_
+    -   the action button/link and the close button will be focusable tab stops
+-   _Form Input_
+    -   N/A
+-   _Use with Assistive Technology_
+    -   the banner will have the ARIA role of `status`. The role `alert` was considered, but it is too aggressive for the range of banner use cases.
+    -   the `status` role has implicit `aria-live` value of `polite`
+    -   if the user supplies a title, we will set `aria-label` to that value
```

Comment:
> if the user supplies a title, we will set `aria-label` to that value
> the user may specify `aria-label` on the `nimble-banner`, but it is not required

Does this mean that the recommendation is to either supply content in their `title` slot or set the _aria-label_ attribute? In general, we try not to require clients to use ARIA attributes directly. Should we take an approach similar to other nimble components where we have a `title-hidden` attribute so that we always tell people to use the `title` slot even if they don't want to display it. That would match `header-hidden` in the dialog and `content-hidden` in the button elements.

Reply:
Yes, let's be consistent. I'll make that change.